websites/R/Radio.net/presence.ts

```diff
@@ -1,187 +1,262 @@
-          const adlength = status.textContent.match(/\d+/g)
-            ? parseInt(status.textContent.match(/\d+/g)[0])
-            : 0;
-
-          if (adlength > 0) {
-            presenceData.state = "Currently watching an ad";
-
-            presenceData.startTimestamp = Math.floor(Date.now() / 1000);
-            presenceData.endTimestamp = presenceData.startTimestamp + adlength;
-          } else presenceData.state = status.textContent;
-        } else {
-          //Player is inactive (no status is being displayed)
-          presenceData.state =
-            document.querySelector<HTMLElement>(".z7kxsz-11").textContent;
-        }
-      }
-      break;
+  const host = window.location.hostname.split("."),
+    path = window.location.pathname.split("/").slice(1);
+
+  if (host[0] === "corporate") {
+    // Corporate page
+    switch (path[0]) {
+      // About us
+      // Broadcasters
+      // Advertising
+      // Press
+      // Jobs
+      // Contact
+      case "about-us":
+      case "ueber-uns":
+      case "broadcasters":
+      case "sender":
+      case "advertising":
+      case "werbung":
+      case "press":
+      case "presse":
+      case "jobs":
+      case "contact":
+      case "kontakt":
+        await handleCorporate(host.slice(1).join("."));
+        break;
+      default:
+        presence.setActivity();
+        break;
+    }
+  } else {
+    // Main page
+    switch (path[0]) {
+      // Radio station
+      case "s":
+        await handleStation();
+        break;
+      // Podcast
+      case "p":
+        await handlePodcast();
+        break;
+      // Search
+      case "search":
+        await handleSearch();
+        break;
+      // Genre
+      // Topic
+      // Country
+      // City
+      // Local stations
+      // Top stations
+      case "genre":
+      case "topic":
+      case "country":
+      case "city":
+      case "local-stations":
+      case "top-stations":
+        await handleListing();
+        break;
+      // Choose your country
+      // Contact
+      // App
+      // Terms and conditions
+      // Privacy policy
+      // Imprint
+      case "country-selector":
+      case "contact":
+      case "app":
+      case "terms-and-conditions":
+      case "privacy-policy":
+      case "imprint":
+        await handleGeneric(true);
+        break;
+      // Startpage
+      // Unknown
+      default:
+        presence.setActivity();
+        break;
     }
-    //Search
-    case "search": {
-      browsingTimestamp = 0;
+  }
+});
+
+/**
+ * Get Language Strings
+ * @returns Language Strings
+ */
+async function getStrings(): Promise<LangStrings> {
+  return presence.getStrings(
+    {
+      play: "general.playing",
+      pause: "general.paused",
+      search: "general.searching",
+      browsing: "general.browsing"
+    },
+    (await presence.getSetting("lang").catch(() => "en")) as string
+  );
+}
+
+/**
+ * Handle radio station
+ */
```

Comment:
I don't see the need of having the doc block which basically reiterates the name of the function, it is obvious what it handles...

Reply:
Consistensy